zerver/tests/test_auth_backends.py

```diff
@@ -94,7 +94,11 @@ def verify_backend(self, backend: Any, good_kwargs: Optional[Dict[str, Any]]=Non
 
         # Verify auth fails with a deactivated user
         do_deactivate_user(user_profile)
-        self.assertIsNone(backend.authenticate(**good_kwargs))
+        if not isinstance(backend, GitHubAuthBackend):
```

Comment:
We can future-proof this by doing SocialAuthMixin here.  

Also, we should have an else clause -- best to just test both cases.